libs/utils/utils.js

```diff
@@ -853,6 +854,13 @@ async function checkForPageMods() {
     persManifests = persManifests.concat(getPromoManifests(promoMd));
   }
 
+  if (dynamicNavMd) {
```

Comment:
I'd do something such as this in the gnav block:
```
 let url = getMetadata('gnav-source') || `${locale.contentRoot}/gnav`;
 if(dynamicNav) {
    const { default: dynamicNav } = await import('../features/dynamic-navigation.js');
   url = getDynamicNavUrl(); // just import getting the config / metadata in the module itself rather than passing them as params
 }
```

Reply:
I will add my feedback here since I cannot add it in the compare URL. @JasonHowellSlavin - the dynamicNav method should only be imported if the key is set, see @mokimo 's code example.